### PR TITLE
Query builder for DynamoDB

### PR DESCRIPTION
**Pull request by obecker**
- provide filter DSL for scan operations
- add unit tests
- amend README

### CODE REVIEW

File: amazon/dynamodb/fake/src/test/kotlin/org/http4k/connect/amazon/dynamodb/mapper/DynamoDbTableMapperFixtures.kt [hunk 1]
```diff
@@ -31,6 +32,7 @@ internal val bandit = Cat(
     ownerId = owner2,
     id = UUID.fromString("c27d8f9d-ffac-45ce-889f-9024d9d0bf2d"),
     name = "Bandit",
+    nickName = "Fatty",
```
Comment (oharaandrew314) [inline amazon/dynamodb/fake/src/test/kotlin/org/http4k/connect/amazon/dynamodb/mapper/DynamoDbTableMapperFixtures.kt:35]: This man knows too much.

**Comment by oharaandrew314** [comment]
Seems 👍 to me

**Comment by obecker** [comment]
Hi @daviddenton, I've seen you added this feature already to the change log in your last commit. Don't forget to merge the branch `dynamodb-query-builder` into `master`. 😀

**Comment by daviddenton** [comment]
ah - good catch! :)

**Pull request merged**